nilearn/regions/region_extractor.py

```diff
@@ -0,0 +1,483 @@
+        score of the input data of the maps.
+
+    Returns
+    -------
+    threshold_maps_img: a Nifti like image/object
+        a thresholded image of the input.
+    """
+    maps_img = check_niimg(maps_img)
+    if len(maps_img.shape) > 3:
+        single_map = False
+    else:
+        single_map = True
+
+    if mask_img is not None:
+        mask_img = check_niimg_3d(mask_img)
+        if parameters is not None:
+            maps, affine = filter_and_mask(maps_img, mask_img, parameters)
+        else:
+            nifti_masker = NiftiMasker(mask_img=mask_img)
+            maps = nifti_masker.fit_transform(maps_img)
+    elif mask_img is None:
+        nifti_masker = NiftiMasker(mask_img=mask_img)
+        maps = nifti_masker.fit_transform(maps_img)
+        mask_img = nifti_masker.mask_img_
+
+    ratio = None
+    if isinstance(threshold, float):
+        if estimate_threshold_value is None:
+            # When threshold is needed to apply directly based on
+            # statistical values which should not be more than maximum
+            value_check = abs(maps).max()
+            if abs(threshold) > value_check:
+                raise ValueError("The value given to threshold "
+                                 "statistical maps must not exceed %d. "
+                                 "You provided threshold=%s " % (value_check,
+                                                                 threshold))
+            else:
+                cutoff_threshold = threshold
+        else:
+            ratio = threshold
+    elif threshold == 'auto':
+        ratio = 0.8
+    elif threshold is not None:
+        raise ValueError("Threshold must be either "
+                         "'auto' or float. You provided %s." %
+                         str(threshold))
+    # check if the input strategy is a valid
+    list_of_estimators = [None, 'percentile', 'ratio_n_voxels']
+    if estimate_threshold_value not in list_of_estimators:
+        message = ("'threshold_strategy' should be given as "
+                   "either of these {0}").format(list_of_estimators)
+        raise ValueError(message)
+
+    # Thresholding
+    if ratio is not None and ratio > 1.:
+        raise ValueError("threshold given for a 'percentile' or 'ratio_n_voxels' "
+                         "is expected between 0. and 1. "
+                         "You provided %s" % threshold)
+
+    if estimate_threshold_value == 'percentile':
+        percentile = 100. - (100. / len(maps)) * ratio
+        cutoff_threshold = scoreatpercentile(np.abs(maps), percentile)
+    elif ratio is not None and estimate_threshold_value == 'ratio_n_voxels':
+        raveled = np.abs(maps).ravel()
+        argsort = np.argsort(raveled)
+        n_voxels = (ratio * maps.size)
+        cutoff_threshold = raveled[argsort[- n_voxels]]
+
+    maps[np.abs(maps) < cutoff_threshold] = 0.
+    threshold_maps = maps
+    threshold_maps_img = masking.unmask(threshold_maps, mask_img)
```

Comment:
I used this just for my comfortability. Other alternatives for example, ```new_img_like``` ? Do you think ?